src/dbnode/server/server.go

```diff
@@ -1165,6 +1175,79 @@ func kvWatchEncodersPerBlockLimit(
 	}()
 }
 
+func kvWatchQueryLimit(
+	store kv.Store,
+	logger *zap.Logger,
+	limit limits.LookbackLimit,
+	kvName string,
+) {
+	options := limit.Options()
+
+	value, err := store.Get(kvName)
+	if err == nil {
+		protoValue := &commonpb.StringProto{}
+		err = value.Unmarshal(protoValue)
+		if err == nil {
+			options = parseLookbackLimitOptions(logger, kvName, protoValue.Value, options)
+		}
+	} else if errors.Is(err, kv.ErrNotFound) {
+		logger.Warn("error resolving query limit", zap.Error(err), zap.String("name", kvName))
+	}
+
+	if err := limit.Update(options); err != nil {
+		logger.Warn("unable to set query limit", zap.Error(err), zap.String("name", kvName))
+	}
+
+	watch, err := store.Watch(kvName)
+	if err != nil {
+		logger.Error("could not watch query limit", zap.Error(err), zap.String("name", kvName))
+		return
+	}
+
+	go func() {
+		protoValue := &commonpb.StringProto{}
+		for range watch.C() {
+			value := options
+			if newValue := watch.Get(); newValue != nil {
+				if err := newValue.Unmarshal(protoValue); err != nil {
+					logger.Warn("unable to parse new query limit", zap.Error(err), zap.String("name", kvName))
+					continue
+				}
+				value = parseLookbackLimitOptions(logger, kvName, protoValue.Value, value)
+			}
+
+			if err := limit.Update(value); err != nil {
+				logger.Warn("unable to set query limit", zap.Error(err), zap.String("name", kvName))
+			}
+		}
+	}()
+}
+
+func parseLookbackLimitOptions(logger *zap.Logger,
+	kvName string,
+	val string,
+	defaultOpts limits.LookbackLimitOptions,
+) limits.LookbackLimitOptions {
+	parts := strings.Split(val, ",")
```

Comment:
May be worth using a slightly more structured style here, eg json encoded limit values

Reply:
Converted to use a single struct w/ all the limits